documentation/templates/privilege_escalation.md.erb

```diff
@@ -99,3 +99,65 @@ This enables using any executable and command-line options to change users. Keep
 - Bolt always appends the specified `run-as` user to the specified `run-as-command`,
   which can be limiting for some escalating commands.
 - Bolt will not use the `run-as-command` option if `run-as` is not set.
+
+## Escalating privilege over WinRM
+
+The WinRM transport does not support `run-as`. This is because WinRM operates
+with a session level token and has no functionality to change to another user
+after authentication. Once you connect with a user, you can only perform
+actions that you have permission to perform with that user.  Instead of
+switching users, you can connect with a user, and then provide credentials to a
+program that accepts them to perform another action. The behavior of providing
+credentials to a program is not part of WinRM.
+
+For example, to get the processes running on a Windows target, you could
+connect to the target with WinRM using `Invoke-BoltScript -Script
+.\myscript.ps1 -Targets winrm://mytarget -User userA -Password passwordA` and
+then use the following Powershell script to pass credentials to the
+`Get-WmiObject` command:
+```
+$Cred = New-Object System.Management.Automation.PsCredential('userB',$passwordB)
+Get-WmiObject -Class Win32_BIOS -Computer SERVER1 -Credential $Cred
+```
+
+If you want to securely pass your userB credentials into a powershell script you have a few options.
+The best way to pass parameters securely into a script is by [converting it to a
+task](writing_tasks.md#converting-scripts-to-tasks) and [defining sensitive
+parameters](writing_tasks.md#defining-sensitive-parameters) for that task. This will obfuscate the
+secret in logs and API responses.
+
+You can pass sensitive parameters to a task directly if your secrets are the same for every target.
+If your secrets are unique to each target, or need to be computed, fetched, or obfuscated when you
+pass them to the task you will want to set your secret as data on the target in the inventory. For
+the purposes of escalating as a secondary user you can set arbitrary variables on targets in the
+inventory, which you can then access.
+
+There are a few ways of setting and retrieving variables on targets. You can set variables on a
```

Comment:
I'm a little confused here. I think it's just cos there's a lot going on in this paragraph and I'm not sure if it's describing an alternate way to do the same thing as the previous paragraph, or an additional thing that you need to do. I'm guessing it's an alternate way based on the summary later.

So far, we've covered using a task, defining sensitive parameters, and passing sensitive parameters into the task using an inventory file or directly. Is this paragraph covering a new use case - using an inventory plugin or Hiera data along with a remote task instead of defining sensitive parameters?

Reply:
I'm just going to remove the Hiera bit. You can definitely use it along these lines, but it's so complicated and I think anyone who isn't familiar with Hiera can just do the inventory workflow